File: bindings/php/src/ProvenSemweb.php
<?php
// SPDX-License-Identifier: PMPL-1.0-or-later
// Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <j.d.a.jewell@open.ac.uk>
//
// Semantic Web protocol types for proven-servers.

declare(strict_types=1);

namespace ProvenServers;

/** RdfFormat matching the Idris2 ABI tags. */
enum RdfFormat: int
{
    case RdfXml = 0;
    case Turtle = 1;
    case NTriples = 2;
    case NQuads = 3;
    case JsonLd = 4;
    case Trig = 5;
}

/** SemwebResourceType matching the Idris2 ABI tags. */
enum SemwebResourceType: int
{
    case Class = 0;
    case Property = 1;
    case Individual = 2;
    case Ontology = 3;
    case NamedGraph = 4;
}

/** HttpMethod matching the Idris2 ABI tags. */
enum HttpMethod: int
{
    case Get = 0;
    case Post = 1;
    case Put = 2;
    case Patch = 3;
    case Delete = 4;
}

/** ContentNegotiation matching the Idris2 ABI tags. */
enum ContentNegotiation: int
{
    case NegRdfXml = 0;
    case NegTurtle = 1;
    case NegJsonLd = 2;
    case NegHtml = 3;
}

/** SemwebErrorCode matching the Idris2 ABI tags. */
enum SemwebErrorCode: int
{
    case NotFound = 0;
    case InvalidUri = 1;
    case MalformedRdf = 2;
    case UnsupportedFormat = 3;
    case ConflictingTriples = 4;
}

File: bindings/csharp/src/ProvenSemweb.cs
// SPDX-License-Identifier: PMPL-1.0-or-later
// Copyright (c) 2026 Jonathan D.A. Jewell (hyperpolymath) <j.d.a.jewell@open.ac.uk>
//
// Semantic Web protocol types for proven-servers.

namespace Proven;

/// <summary>RdfFormat matching the Idris2 ABI tags (0-5).</summary>
public enum RdfFormat : byte
{
    RdfXml = 0,
    Turtle = 1,
    NTriples = 2,
    NQuads = 3,
    JsonLd = 4,
    Trig = 5
}

/// <summary>SemwebResourceType matching the Idris2 ABI tags (0-4).</summary>
public enum SemwebResourceType : byte
{
    Class = 0,
    Property = 1,
    Individual = 2,
    Ontology = 3,
    NamedGraph = 4
}

/// <summary>HttpMethod matching the Idris2 ABI tags (0-4).</summary>
public enum HttpMethod : byte
{
    Get = 0,
    Post = 1,
    Put = 2,
    Patch = 3,
    Delete = 4
}

/// <summary>ContentNegotiation matching the Idris2 ABI tags (0-3).</summary>
public enum ContentNegotiation : byte
{
    NegRdfXml = 0,
    NegTurtle = 1,
    NegJsonLd = 2,
    NegHtml = 3
}

/// <summary>SemwebErrorCode matching the Idris2 ABI tags (0-4).</summary>
public enum SemwebErrorCode : byte
{
    NotFound = 0,
    InvalidUri = 1,
    MalformedRdf = 2,
    UnsupportedFormat = 3,
    ConflictingTriples = 4
}
